controllers/controllers/workloads/cfspace_controller.go

```diff
@@ -127,28 +110,24 @@ func (r *CFSpaceReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ct
 
 	err = propagateSecrets(ctx, r.client, r.log, cfSpace, r.packageRegistrySecretName)
 	if err != nil {
-		r.log.Error(err, fmt.Sprintf("Error propagating secrets into CFSpace %s/%s", req.Namespace, req.Name))
+		r.log.Error(err, fmt.Sprintf("Error propagating secrets into CFSpace %s/%s", cfSpace.Namespace, cfSpace.Name))
 		return ctrl.Result{}, err
 	}
 
 	err = reconcileRoleBindings(ctx, r.client, r.log, cfSpace)
 	if err != nil {
-		r.log.Error(err, fmt.Sprintf("Error propagating role-bindings into CFSpace %s/%s", req.Namespace, req.Name))
+		r.log.Error(err, fmt.Sprintf("Error propagating role-bindings into CFSpace %s/%s", cfSpace.Namespace, cfSpace.Name))
 		return ctrl.Result{}, err
 	}
 
 	err = r.reconcileServiceAccounts(ctx, cfSpace)
 	if err != nil {
-		r.log.Error(err, fmt.Sprintf("Error propagating service accounts into CFSpace %s/%s", req.Namespace, req.Name))
+		r.log.Error(err, fmt.Sprintf("Error propagating service accounts into CFSpace %s/%s", cfSpace.Namespace, cfSpace.Name))
 		return ctrl.Result{}, err
 	}
 
 	cfSpace.Status.GUID = namespace.Name
-	err = updateStatus(ctx, r.client, cfSpace, metav1.ConditionTrue)
-	if err != nil {
-		r.log.Error(err, fmt.Sprintf("Error updating status on CFSpace %s/%s", req.Namespace, req.Name))
-		return ctrl.Result{}, err
-	}
+	setStatusCondition(&cfSpace.Status.Conditions, StatusConditionReady, metav1.ConditionTrue, StatusConditionReady, "")
```

Comment:
It is arguable if this helper func is more readable than the `meta.SetStatusCondition` utility it wraps, where the literal syntax makes the meaning of each string more clear. Maybe consider inlining?

Reply:
Yeah, I was thinking about the same. I tried replacing the function with `meta.SetStatusCondition` on one place and I thought that looks a bit more unreadable. Can't really make up my mind, but as you raised the question as well, I guess it is worth doing.